src/ring_theory/polynomial/hermite.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+-/
+
+import data.polynomial.derivative
+
+/-!
+# Hermite polynomials
+
+This file defines `polynomial.hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `polynomial.hermite n`: the `n`th probabilist's Hermite polynomial,
+  defined recursively as a `polynomial ℤ`
+
+## References
+
+* [Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+namespace polynomial
+
+/-- the nth probabilist's Hermite polynomial -/
+noncomputable def hermite : ℕ → polynomial ℤ
+| 0     := 1
+| (n+1) := X * (hermite n) - (hermite n).derivative
+
+@[simp] lemma hermite_succ (n : ℕ) : hermite (n+1) = X * (hermite n) - (hermite n).derivative :=
+by rw hermite
+
+lemma hermite_eq_iterate (n : ℕ) : hermite n = ((λ p, X*p - p.derivative)^[n] 1) :=
```

Comment:
```suggestion
lemma hermite_eq_iterate (n : ℕ) : hermite n = (λ p, X*p - p.derivative)^[n] 1 :=
```

Reply:
When I remove those brackets, I get an error. The error message with the cursor placed right before the `=` is:
```
type mismatch at application
  hermite n = λ (p : polynomial ?m_1), X * p - ⇑derivative p
term
  λ (p : polynomial ?m_1), X * p - ⇑derivative p
has type
  polynomial ?m_1 → polynomial ?m_1 : Type ?
but is expected to have type
  polynomial ℤ : Type
Additional information:
...\mathlib_hermite\src\ring_theory\polynomial\hermite.lean:38:45: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  type mismatch, term
    ?m_2 = ?m_3
  has type
    Prop : Type
  but is expected to have type
    ?m_1 → ?m_1 : Sort ?
```
and the error message with the cursor placed right before the `1` is:
```
failed to synthesize type class instance for
n : ℕ
⊢ has_one (Sort ?)
```

Reply:
It seems the notation for `nat.iterate` has lower precedence than `=`, which is very weird. Let's keep the parentheses then and hope that we can remove them in mathlib4.